From 0dbafc2185757ed18d30555a8d0b2fab924ad63d Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Fri, 30 Mar 2012 11:48:00 +0200
Subject: [PATCH 3/3] usb-ehci: drop assert()

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1333108080-16101-1-git-send-email-kraxel@redhat.com>
Patchwork-id: 39039
O-Subject: [RHEL-6.3 qemu-kvm PATCH] usb-ehci: drop assert()
Bugzilla: 807916
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

Not sure what the purpose of the assert() was, in any case it is bogous.
We can arrive there if transfer descriptors passed to us from the guest
failed to pass sanity checks, i.e. it is guest-triggerable.  We deal
with that case by resetting the host controller.  Everything is ok, no
need to throw a core dump here.

bugzilla: 807916 - boot from the USB storage core dumped after press
                   "ctrl-alt-delete"
upstream: http://patchwork.ozlabs.org/patch/149629/
brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=4236816

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
---
 hw/usb-ehci.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

Signed-off-by: Michal Novotny <mignov@gmail.com>
---
 hw/usb-ehci.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/hw/usb-ehci.c b/hw/usb-ehci.c
index 59f8a53..7ab0080 100644
--- a/hw/usb-ehci.c
+++ b/hw/usb-ehci.c
@@ -2003,7 +2003,6 @@ static void ehci_advance_state(EHCIState *ehci,
             fprintf(stderr, "processing error - resetting ehci HC\n");
             ehci_reset(ehci);
             again = 0;
-            assert(0);
         }
     }
     while (again);
-- 
1.7.7.6

